template/docker-entrypoint.sh

```diff
@@ -245,7 +245,7 @@ docker_setup_db() {
 	fi
 	# Generate random root password
 	if [ -n "$MYSQL_RANDOM_ROOT_PASSWORD" ]; then
-		export MYSQL_ROOT_PASSWORD="$(pwgen -1 32)"
+		MYSQL_ROOT_PASSWORD="$(openssl rand -hex 16)"; export MYSQL_ROOT_PASSWORD
```

Comment:
This is _technically_ a pretty major decrease in bits of entropy, but I'm not sure whether it's important.

The current `pwgen` is very roughly somewhere around (26×2+10)^32 = ~2.3×10^57, where this is 8^16 = ~2.8×10^14.

This is essentially the same as MD5 -- if we're worried about it, we could double it to 32 and get to the same length as a SHA256, but I'm not sure it makes much difference? (also want to be wary of MySQL's password length limits, which I seem to recall being something close to 32 characters?)

We could also increase our entry via something like base64'd random data instead (and trim it to the appropriate length), ala something like `openssl rand -base64 32 | cut -b -32`.

Reply:
No. I think those symbols should be fine (might actually be a requirement to have some special symbols if users enable the password_validation plugin).

Reply:
openssl rand -base64 isn't guaranteed to include special characters required by the password_validation plugin. However, that plugin isn't enabled by default, and pwgen as used in the Debian-based images never includes special characters, so I would suggest at most making a little note about it in the README.